src/Blocks/BlockDoor.cpp

```diff
@@ -50,10 +50,20 @@ void cBlockDoorHandler::OnUse(cChunkInterface & a_ChunkInterface, cWorldInterfac
 	UNUSED(a_CursorY);
 	UNUSED(a_CursorZ);
 
-	if (a_ChunkInterface.GetBlock(a_BlockX, a_BlockY, a_BlockZ) == E_BLOCK_WOODEN_DOOR)
+	switch (a_ChunkInterface.GetBlock(a_BlockX, a_BlockY, a_BlockZ))
```

Comment:
With an `ASSERT(!"Unhandled door block type");` in it so that it is quickly found when such a case happens.